sktime/transformations/series/adapt.py

```diff
@@ -196,7 +200,8 @@ def _get_args(self, X, y, method="fit"):
 
         The return is a dict which is passed to the method of name method.
         """
-        if not self._trafo_has_param_and_default(method, "X"):
+        trafo_has_X = self._trafo_has_param_and_default(method, "X")[0]
+        if not trafo_has_X:
             return {"y": X}
```

Comment:
Just a note: if you are familiar with walrus operator (:=), this type of assignment and immediate reuse gets a bit easier in one line. I love that operator, so just thought about sharing.

https://docs.python.org/3/whatsnew/3.8.html#assignment-expressions

Reply:
matter of taste, but tbh I do not like it, since it mixes assignment with check. It makes things much harder to read.

I tend to err on the side of easier to understand code when multiple options are available.

Another example is, at the start of a method in a class, doing things like `varname = self.varname`, `var2 = self.var2`, etc, and then not using `self` again in the method body, so it is clear to the reader with `self` attrs the function accesses. Even if - in the function body - using `self.var2` everywhere instead of `var2` might be more compact.